Feature/container sell value

FluxCapacitor2 opened the pull request:
I added an overlay that renders when a chest, ender chest page, backpack, or minion is open and shows the container's 20 most valuable items along with the combined value of all items in the container. It can be moved and scaled just like a regular `GuiElement`, but it is rendered after the container background is drawn to prevent the text from being dimmed (similar to the `ChestProfit` element). When the overlay is rendered over a minion GUI, minion upgrades and fuels aren't included in the value.

* Should I add config options to control the maximum amount of items to display?
* Should I include things like enchantments and HPBs in item prices?
* Should enchanted books be grouped by enchantments or by item name? Currently all items are grouped by item name, and enchanted books are only displayed as "Enchanted Book" instead of something like "Soul Eater I"
* Let me know if you want any formatting changes

Here's an example of this overlay being used while inside a minion GUI:
![image](https://user-images.githubusercontent.com/31071265/159999927-0ad341a9-53d2-4048-9460-8f24d74a8731.png)

Example of the truncation:
![image](https://user-images.githubusercontent.com/31071265/160008085-13d3caa2-51c8-4a5d-98bf-af8c347d2bb0.png)

FluxCapacitor2 committed:
Container sell value

FluxCapacitor2 committed:
Fix incorrect prices for enchanted books

Lines in the display were stacked by their item name, and instead of keeping track of the `ItemStack`s in a list, I just incremented a variable showing the amount of items that were compacted into that line. This has now been fixed.

FluxCapacitor2 committed:
Fix text alignment when the element is near the edge of the screen

FluxCapacitor2 committed:
Fix showing the overlay when there were no items in a container

FluxCapacitor2 committed:
Fix all the things that Codacy is yelling at me for (hopefully)

Sychic commented on src/main/kotlin/skytils/skytilsmod/features/impl/misc/ContainerSellValue.kt:
```diff
@@ -0,0 +1,193 @@
+ * and backpacks that shows the top items in the container by value and the container's total value.
+ * @author FluxCapacitor2
+ */
+class ContainerSellValue {
```
Since this is a singleton, it should be an object

Sychic replied:
```suggestion
object ContainerSellValue {
```

Sychic commented on src/main/kotlin/skytils/skytilsmod/features/impl/misc/ContainerSellValue.kt:
```diff
@@ -0,0 +1,193 @@
+    companion object {
+        val element = SellValueDisplay()
+        // The max amount of BIN prices to show in the list. If there are more items, the list will be truncated to this size.
+        const val MAX_DISPLAYED_ITEMS = 20
```
maybe this should be a setting

Sychic commented on src/main/kotlin/skytils/skytilsmod/features/impl/misc/ContainerSellValue.kt:
```diff
@@ -0,0 +1,193 @@
+        textLines.add("§eTotal Value: §a${ NumberUtil.format(totalContainerValue) }")
+
+        // Translate and scale manually because we're not rendering inside the GuiElement#render() method
+        GlStateManager.pushMatrix()
+        GlStateManager.translate(element.actualX, element.actualY, 0f)
+        GlStateManager.scale(element.scale, element.scale, 0f)
+
+        textLines.forEachIndexed { i, str ->
+            ScreenRenderer.fontRenderer.drawString(
+                str,
+                element.textPosX, (i * ScreenRenderer.fontRenderer.FONT_HEIGHT).toFloat(),
+                CommonColors.WHITE, element.alignment, SmartFontRenderer.TextShadow.NORMAL
+            )
+        }
+
+        GlStateManager.popMatrix()
```
Ideally this should be using UMatrixStack

Sychic commented:
Enchants would definitely make more sense as the enchant rather than just `Enchanted Book`

FluxCapacitor2 committed:
Various improvements

- Converted the class to an object because it is a singleton
- Added config option for the maximum number of displayed items
- Made enchanted books show their enchant name instead of just "Enchanted Book"
- Convert regular GL transformations into a `UMatrixStack`
- Fix truncation showing an incorrect number of extra items
- Fix text alignment (finally)

FluxCapacitor2 committed:
Why do you do this to me Codacy

FluxCapacitor2 committed:
Add dependency in vigilance config

FluxCapacitor2 commented:
Fixed, enchants now look like this:
![image](https://user-images.githubusercontent.com/31071265/160708762-efedd64e-0fab-4830-9e35-dc353876b21a.png)
Enchanted books with only one enchant use the first line of their lore to determine which string to display, and if a book has multiple enchants, it is still labeled as "Enchanted Book"

FluxCapacitor2 committed:
Reformat
